lark/parse_tree_builder.py

```diff
@@ -30,23 +30,36 @@ def __init__(self, node_builder, node_filter=None):
     def __call__(self, children):
         res = self.node_builder(children)
 
-        # local reference to Tree.meta reduces number of presence checks
         if isinstance(res, Tree):
+            # Calculate positions while the tree is streaming, according to the rule:
+            # - nodes start at the start of their first child's container,
+            #   and end at the end of their last child's container.
+            # Containers are nodes that take up space in text, but have been inlined in the tree.
+
             res_meta = res.meta
 
             first_meta = self._pp_get_meta(children)
             if first_meta is not None:
-                res_meta.line = first_meta.line
-                res_meta.column = first_meta.column
-                res_meta.start_pos = first_meta.start_pos
-                res_meta.empty = False
+                if not hasattr(res_meta, 'line'):
+                    # meta was already set, probably because the rule has been inlined (e.g. `?rule`)
```

Comment:
Can you create a test showcasing this?

Reply:
Added.

Reply:
@erezsh Could you approve my request to use Jetbrains (e.g. my IDE) to access this repo?

Reply:
Approved.

Reply:
Thank you.